src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/formatWriter/ByteRecordWriterProvider.java

```diff
@@ -0,0 +1,59 @@
+package com.microsoft.azure.kusto.kafka.connect.sink.formatWriter;
+
+import com.microsoft.azure.kusto.kafka.connect.sink.format.RecordWriter;
+import com.microsoft.azure.kusto.kafka.connect.sink.format.RecordWriterProvider;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+
+public class ByteRecordWriterProvider implements RecordWriterProvider {
+  private static final Logger log = LoggerFactory.getLogger(ByteRecordWriterProvider.class);
+
+  @Override
+  public RecordWriter getRecordWriter(String filename, OutputStream out) {
+    return new RecordWriter() {
+      long size =0;
+
+      @Override
+      public void write(SinkRecord record) throws IOException {
+        byte[] value = null;
+        byte[] valueBytes = (byte[]) record.value();
+        byte[] separator = "\n".getBytes(StandardCharsets.UTF_8);
```

Comment:
as discussed - we should drop this for binary formats

Reply:
@ohadbitt I guess we are still supporting Avro and JSON as bytes. 
cc @anagha-microsoft

Reply:
yea i meant just the separator addition ... this should be behind format check - i know its working for avro but it might not work for other binary formats